### DISCUSSION TITLE
Can I create an Image Reconstruction Deep Neural Network?

### DISCUSSION THREAD

**Discussion by alphastint** [davisking/dlib#2809] [General]
I would like to implement Variational Auto Enconding algorithm. The core of the algorithm is that it compresses the database into a two dimensional vector, than it tries to reconstruct the original image from the two parameters. 
If I have an n x n x rgb image I would like a network like n x n x rgb -> 2 -> n x n x rgb. The loss function would be the norm between the original and the reconstructed image. I have tried to use

`loss_mean_squared_per_channel_and_pixel`

but with both the input and the label being the same image, but it does not compile. 

Is there an example in dlib for image to image metric?

Thanks in advance.

**Comment by arrufat** [comment]
Without further information, it's impossible to tell you what's wrong.
Check out this example in the dlib's test code to see how to use that loss and how to set up the labels: https://github.com/davisking/dlib/blob/18927f93e223fa2cc1110dec9467949b5605169e/dlib/test/dnn.cpp#L2874-L2947
